### PR TITLE
Move TODO regarding pickling to markdown

### PR DESCRIPTION
**Pull request by lgeiger**
## Change Summary

As far as I can tell pickling is supported by pydantic, so I think this TODO in the docs can be removed.

## Checklist

* [x] The pull request title is a good summary of the changes - it will be used in the changelog
* [x] Unit tests for the changes exist
* [x] Tests pass on CI
* [x] Documentation reflects the changes where applicable
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @adriangb

### CODE REVIEW

**Comment by lgeiger** [comment]
please review

File: docs/concepts/serialization.md [hunk 1]
```diff
@@ -642,7 +642,6 @@ print(u.model_dump_json())  # (1)!
 Pydantic models support efficient pickling and unpickling.
 
 ```py test="skip"
```
Comment (adriangb) [inline docs/concepts/serialization.md:644]:
```suggestion
```py
```

**Comment by adriangb** [comment]
We'll have to get the test working to verify that!

**Comment by lgeiger** [comment]
> We'll have to get the test working to verify that!

Sorry about that, I re-enabled the test. At least locally it seems to work as expected.

**Review by sydney-runkle — APPROVED** [review]
Great, thank you!

**Comment by sydney-runkle** [comment]
Oh hmm, looks like this is still failing tests:

```
data = pickle.dumps(m)
E   _pickle.PicklingError: Can't pickle <class '__main__.FooBarModel'>: attribute lookup FooBarModel on __main__ failed
```

**Comment by lgeiger** [comment]
Yeah, looks like the failure just happens in the doctests. I moved the comment to a markdown such that it doesn't show up in the public docs which might confuse some users

**Comment by sydney-runkle** [comment]
Ah nice. Looks good!

**Pull request merged**